src/NCronJob/Registry/JobHistory.cs

```diff
@@ -2,11 +2,26 @@
 
 namespace NCronJob;
 
-internal sealed class JobHistory
+internal interface IJobHistory
 {
+    void Add(JobRun jobRun);
+    IReadOnlyCollection<JobRun> GetAll();
+}
+
+internal sealed class JobHistory : IJobHistory
+{
+    //Note: This can grow out of control if not managed properly. JobHistory should ultimately be persisted to a database or something durable.
```

Comment:
Good point - as you said some time ago, we might wanna invest some time into **FASTER**.

We could also have a `FixedSizeBag` with an upper limit.

Reply:
Right, the idea is that the Write-Ahead Log from FASTER (or equivalent) should ensure data integrity even in the event of a crash. This can also help with rehydrating the data back when the process is restored.

Reply:
Exactly - probably configurable.

My current thinking is that jobs (by default) shouldn't re-run - but the user explicitly has to ask "us"